app/controllers/WKRemoteDataStoreController.scala

```diff
@@ -189,11 +191,16 @@ class WKRemoteDataStoreController @Inject()(
         existingDataset = datasetDAO
           .findOneByNameAndOrganizationName(datasourceId.name, datasourceId.team)(GlobalAccessContext)
           .futureBox
+
         _ <- existingDataset.flatMap {
           case Full(dataset) =>
-            datasetDAO
-              .deleteDataset(dataset._id)
-              .flatMap(_ => usedStorageService.refreshStorageReportForDataset(dataset))
+            for {
+              annotations <- annotationDAO.findAllByDataset(dataset._id)(GlobalAccessContext)
```

Comment:
```suggestion
              annotations <- annotationDAO.countAllByDataset(dataset._id)(GlobalAccessContext)
```
No need to create objects for all the relevant annotations, if the only operation then is nonEmpty